beater/server_test.go

```diff
@@ -74,12 +74,28 @@ func TestServerFrontendSwitch(t *testing.T) {
 	apm.Handler.ServeHTTP(rec, req)
 	assert.Equal(t, 403, rec.Code, rec.Body.String())
 
-	apm.Handler = newMuxer(Config{EnableFrontend: true}, nil)
+	apm.Handler = newMuxer(Config{EnableFrontend: true, AllowOrigins: []string{"*"}}, nil)
 	rec = httptest.NewRecorder()
 	apm.Handler.ServeHTTP(rec, req)
 	assert.NotEqual(t, 403, rec.Code, rec.Body.String())
 }
 
+func TestServerCORS(t *testing.T) {
+	apm, teardown := setupServer(t, noSSL)
+	defer teardown()
+
+	req, _ := http.NewRequest("POST", FrontendTransactionsURL, bytes.NewReader(testData))
+
+	apm.Handler = newMuxer(Config{EnableFrontend: true,
+		AllowOrigins: []string{"http://notmydomain.com", "http://neitherthisone.com"}},
+		nil)
+
+	rec := httptest.NewRecorder()
+	apm.Handler.ServeHTTP(rec, req)
+
+	assert.Equal(t, 403, rec.Code, rec.Body.String())
```

Comment:
This is testing that if we respond with 403 if there's no `Origin` set, AFAICT. Seems like that's already tested by the Python code in `test_no_origin`.

Reply:
The python tests test against a running apm-server instance. Those tests should be rather high level as they usually take longer to run. 
I think testing that all headers are set properly and testing through all the different combinations of cors handling should be test in unit tests.